app/utils/CloudWatchClient.scala

```diff
@@ -30,8 +28,8 @@ class CloudWatchClient(stage: String) extends Loggable {
     val request = new PutMetricDataRequest().withNamespace("Typerighter").withMetricData(datum)
 
     try {
-      val result = cloudWatchClient.putMetricData(request)
-      log.info(s"Published metric data: $result")
+      cloudWatchClient.map(_.putMetricData(request))
+      log.info(s"Published $metric metric data. ${if (dryRun) "DRY RUN" else ""}")
     } catch {
       case e: Exception =>
         log.error(s"CloudWatch putMetricData exception message: ${e.getMessage}", e)
```

Comment:
Apologies if I am misunderstanding this - but won't this exception always be triggered on a dry run, because the cloudwatchClient will be None? How does the log.info for dry run get triggered if there is no cloudwatchClient for the putMetricData?

Reply:
No exception will be thrown if `cloudWatchClient` is `None` because the `map` function will only map over a `Some` value. This means that we will only call `putMetricData` if we have a `cloudWatchClient`. The message should get logged if there is no `cloudWatchClient` or there is one and it's `put` successfully.

Reply:
Got it! Thanks for explaining